### PR TITLE
refactor rudr run to rudr <workload>:run

### PR DESCRIPTION
**Pull request by wonderflow**

### CODE REVIEW

File: api/types/template_types.go [hunk 1]
```diff
@@ -14,30 +14,31 @@ See the License for the specific language governing permissions and
...
-// TemplateSpec defines the desired state of Template
+// Template defines the content of a plugin
 type Template struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-	Alias            string                    `json:"alias,omitempty"`
-	Object           unstructured.Unstructured `json:"object,omitempty"`
-	LastCommandParam string                    `json:"lastCommandParam,omitempty"`
-	Parameters       []Parameter               `json:"parameters,omitempty"`
+	Name       string                 `json:"name"`
```
Comment (zzxwill) [inline api/types/template_types.go:28]: Is template name the definition name?

File: api/types/template_types.go [hunk 2]
```diff
@@ -14,30 +14,31 @@ See the License for the specific language governing permissions and
...
+	Object     map[string]interface{} `json:"object,omitempty"`
+	Parameters []Parameter            `json:"parameters,omitempty"`
 }
 
+type DefinitionType string
+
+const (
+	TypeWorkload DefinitionType = "workload"
+	TypeTrait    DefinitionType = "trait"
+)
```
Comment (zzxwill) [inline api/types/template_types.go:40]: We can add scope here later as we added the spec.extension for ScopeDefinition in OAM kubernetes runtime.

File: api/types/template_types.go (hunk 1 above)
Comment (wonderflow) [reply api/types/template_types.go:28]: yes

File: api/types/template_types.go (hunk 2 above)
Comment (wonderflow) [reply api/types/template_types.go:40]: yes

**Pull request merged**